SD-K9/src/app/providers/pathfinder.service.spec.ts

```diff
@@ -1,47 +1,123 @@
+                { id: 'node-6', x: 2, y: 8, building: 'na', floor: 0 },
+                { id: 'node-7', x: 5, y: 8, building: 'na', floor: 0 },
+                { id: 'B', x: 8, y: 8, building: 'na', floor: 0 }
+            ];
+        }
+    };
+
+    const nodeA: SVGCoordinate = {
+        id: 'A',
+        building: 'na',
+        floor: 0,
+        x: 2,
+        y: 2,
+    };
+
+    const nodeB: SVGCoordinate = {
+        id: 'B',
+        building: 'na',
+        floor: 0,
+        x: 8,
+        y: 8,
+    };
+
+    /* The walkable nodes look like the following,
+        where the shortest path would be the coordinates marked by *
+
+    (2,2)* (5,2) (8,2)
+    (2,5) (5,5)* (8,5)
+    (2,8) (5,8) (8,8)*
+
+    */
 
     beforeEach(() => {
         TestBed.configureTestingModule({
-            imports: [HttpClientModule],
+            imports: [HttpClientTestingModule],
             providers: [
-                Pathfinder
+                Pathfinder,
+                { provide: SVGManager, useValue: mockSVGManager }
             ]
         }).compileComponents();
         service = TestBed.get(Pathfinder);
     });
 
     afterEach(() => {
         service = null;
-      });
+    });
 
     it('should be created', () => {
         expect(service).toBeTruthy();
     });
 
-    it('#distance should get distance between 2 nodes', () => {
-        const nodeA: SVGCoordinate = {
-            id: 'A',
-            building: 'test',
-            floor: 1,
-            x: 0,
-            y: 0,
-        };
-        const nodeB: SVGCoordinate = {
-            id: 'B',
-            building: 'test',
-            floor: 1,
-            x: 3,
-            y: 4,
+    it('#getShortestPath should get the shortest path between 2 indoor locations', async () => {
+
+        const middleNode: SVGCoordinate = {
+            id: 'node-4',
+            building: 'na',
+            floor: 0,
+            x: 5,
+            y: 5
         };
+
+        const path = await service.getShortestPath(nodeA, nodeB);
+
+        expect(path.length).toEqual(3);
+        expect(path).toEqual([nodeA, middleNode, nodeB]);
+    });
+
+    describe('#getShortestPath should return an error if not on the same floor', () => {
+        const newNodeA = JSON.parse(JSON.stringify(nodeA));
+        newNodeA.floor = 1;
+
+        let resolved: boolean;
+        let rejected: boolean;
+
+        beforeEach((done) => {
+            resolved = false;
+            rejected = false;
+
+            service.getShortestPath(newNodeA, nodeB).then((nodes) => {
+                resolved = true;
+                done();
+            }).catch(() => {
+                rejected = true;
+                done();
+            });
+        });
+
+        it('should reject promise', () => {
+            expect(resolved).toEqual(false);
+            expect(rejected).toEqual(true);
+        });
+    });
+
+    it('#distance should get distance between 2 nodes', () => {
```

Comment:
Since your expectation of the distance is to be equal to eucledian distance, can you update the name to be 
```suggestion
    it('#distance should get eucledian distance between 2 nodes', () => {
```